test/surf-extraction.Tests.ps1

```diff
@@ -66,6 +76,51 @@ The file supplied is not a valid extractor configuration file.
         Get-Extractors | Format-MultilineOutput | Should -Not -Match $extractorName
     }
 
+    It "should be created from a list of module IDs " {
+        New-ExtractorFromModules $extractorName "waives.name" "waives.date" | Format-MultilineOutput | Should -Be @"
+Creating extractor '$extractorName'... [OK]
+"@
+        $LASTEXITCODE | Should -Be 0
+
+        # Verify
+        Get-Extractors | Format-MultilineOutput | Should -Match $extractorName
+    }
+
+    It "should be created from a template" {
+        $extractorTemplate = (Join-Path $PSScriptRoot "extractor-template.json")
+        New-ExtractorFromTemplate $extractorName $extractorTemplate | Format-MultilineOutput | Should -Be @"
+Creating extractor '$extractorName'... [OK]
+"@
+        $LASTEXITCODE | Should -Be 0
+
+        # Verify
+        Get-Extractors | Format-MultilineOutput | Should -Match $extractorName
+    }
+
+    It "should list available modules" {
+        $output = Invoke-App list modules 2>&1 | Format-MultilineOutput
+
+        $output | Should -Match "Name\s+ID\s+Description"
```

Comment:
Note that this should now be "Summary" rather than "Description" as #113 has been merged.